cicd/README.md

```diff
@@ -0,0 +1,55 @@
+# Javabuilder CI/CD
+
+Objectives:
+
+* General
+  * Provide fast feedback on Pull Requests of any build/verify failures
+  * Run all integration/acceptance tests in a non-prod environment
+  * Provide a non-prod development environment that local resources can interact with
+  * Easily create an ad-hoc environment for a given branch, which updates with changes
+  * Eliminate or minimize manual AWS Console configuration
+  * Use policies and roles with least-privilege
+* **CodeBuild** For every PR created that merges into `main`
+  * Build the project, ensure it compiles/transpiles/validates
+  * Run all tests
+  * Update the PR with the build/test status
+  * Provide easily accessible logs and artifacts from the build/test job
+* **CodePipeline** For every commit merged to `main`
+  * Run tests and build artifacts necessary for deployment
+  * Deploy the application to a Test environment
+    * As prod-like as possible
+  * Run integration and acceptance tests in the Test environment
+  * Make test results and artifacts easily accessible to developers
+  * If Test environment passes all tests, then deploy to Prod environment
+  * Deploy the application to a Dev environment
+
+## CI/CD Implementation
+
+There are three phases of the CI/CD configuration. They are best explained in reverse order from the way they flow in practice.
+
+### 3 - App
+
+All of the application resources are defined in a CloudFormation template. We could manually update the CloudFormation stack with an updated template every time we have a code change, but it is considerably less effort and less risky to automate this with a CI/CD pipeline. By using parameters, we can use the same code to deploy multiple environments.
+
+**Note: The application template has not been incorporated into the CI/CD implementation yet. That is the next step. The plan is to migrate 'template.yml.erb' from the root, into this folder.**
+
+We also keep some CodeBuild configuration here, as this code tends to be more coupled to current application code, than the CI/CD code in the next section.
+
+* "app.template.yml" - AWS resources for the application
```

Comment:
"app.template.yml" - Is this just template.yml right now? Will this be renamed as part of this work?

Reply:
Since it's one of several cloudformation templates, I think a specific name can help, TBD whether 'app' is the most appropriate.

Reply:
But yes, this is "template.yml" today.

Reply:
👍